### PR TITLE
Mappers

### PR DESCRIPTION
**Pull request by AinaArd**

### CODE REVIEW

File: src/main/java/ru/itis/services/WishListService.java [hunk 1]
```diff
@@ -21,13 +21,14 @@ public WishListService(WishListRepository wishListRepository, UserService userSe
...
 
     public WishList addNewWishList(String title, String token) {
-        User currentUser = userService.findUserByToken(token).get();
-        WishList newWL = WishList.builder()
-                .title(title)
-                .author(currentUser)
-                .build();
-        currentUser.getWishLists().add(newWL);
-        return wishListRepository.save(newWL);
+        if (userService.findUserByToken(token).isPresent()) {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: Rewrite this method using guard expression.
```java
if (!user.present()) { return defaultWishList(); }
new WishList();
// logic
```

File: src/main/java/ru/itis/services/WishListService.java [hunk 2]
```diff
@@ -21,13 +21,14 @@ public WishListService(WishListRepository wishListRepository, UserService userSe
...
+            User currentUser = userService.findUserByToken(token).get();
+            WishList newWL = new WishList(title, currentUser);
+            currentUser.getWishLists().add(newWL);
+            wishListRepository.save(newWL);
+            return newWL;
+        }
+        return WishList.getDefault();
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: I don't think the default is appropriate in this case.

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 3]
```diff
@@ -25,7 +25,8 @@ public WishListController(ItemService itemService, WishListService wishListServi
     @GetMapping("/lists/{list-id}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Get a single wish list")
-    public WishList getWishList(@PathVariable("list-id") Long listId, @RequestHeader(name = "AUTH") String token) {
+    public WishList getWishList(@PathVariable("list-id") Long listId, @RequestHeader(name = "Authorization")
+            String token) {
 //      TODO: if anonym --> ability to reserve an item
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: TODO thing, fix it or remove it.

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 4]
```diff
@@ -25,7 +25,8 @@ public WishListController(ItemService itemService, WishListService wishListServi
     @GetMapping("/lists/{list-id}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Get a single wish list")
-    public WishList getWishList(@PathVariable("list-id") Long listId, @RequestHeader(name = "AUTH") String token) {
+    public WishList getWishList(@PathVariable("list-id") Long listId, @RequestHeader(name = "Authorization")
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: Let me help you
```java
@GetMapping("/lists/{listId}")
public WishList getWishList(@PathVariable Long listId, ...)
```
Now it is shorter by 12 symbols.

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 5]
```diff
@@ -35,16 +36,17 @@ public WishList getWishList(@PathVariable("list-id") Long listId, @RequestHeader
     @PostMapping("/lists/{list-id}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Add new item to wish list")
-    public void addNewItem(@PathVariable("list-id") Long listId, ItemForm itemForm, @RequestHeader(name = "AUTH") String token) {
-        itemService.addNewItem(itemForm, listId);
+    public void addNewItem(@PathVariable("list-id") Long listId, ItemDto itemDto, @RequestHeader(name = "Authorization")
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: Parameter list may be too long.

File: src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java [hunk 6]
```diff
@@ -8,28 +8,21 @@
 import java.io.IOException;
 
 public class TokenAuthenticationFilter implements Filter {
-    private final static String AUTH_HEADER = "AUTH";
+    private final static String AUTH_HEADER = "Authorization";
```
Comment (Dimanaux) [inline src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java]: You don't use this variable, drop it.

File: src/main/java/ru/itis/services/ItemService.java [hunk 7]
```diff
@@ -21,14 +21,14 @@ public ItemService(ItemsRepository itemsRepository, WishListService wishListServ
...
     }
 
-    public void addNewItem(ItemForm itemForm, Long listId) {
+    public void addNewItem(ItemDto itemDto, Long listId) {
         WishList wishList = getWishList(listId);
         Item newItem = Item.builder()
-                .name(itemForm.getName())
-                .price(itemForm.getPrice())
-                .link(itemForm.getLink())
+                .name(itemDto.getName())
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/ItemService.java]: Too many params. Builder is an antipattern that hides ugly constructor call from you.
If you don't use builder, you will see that your real code
```java
Item newItem = new Item(itemDto.getName(), itemDto.getPrice(), itemDto.getLink(), wishList, itemDto.getDescription())
```
is a mess.

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 5 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: fixed

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 4 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: fixed, thnx

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 5 above)
Comment (Dimanaux) [reply src/main/java/ru/itis/controllers/WishListController.java]: No, there are still 3 parameters.

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 4 above)
Comment (Dimanaux) [reply src/main/java/ru/itis/controllers/WishListController.java]: I meant other thing, please keep this stuff
```
@PreAuthorize("isAuthenticated()")
@ApiOperation("Get a single wish list")
```

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 5 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: if I remove Authorozation header, user won't be able to authorize

Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: maybe it's time to switch to jwt tokens?

File: src/main/java/ru/itis/services/ItemService.java (hunk 7 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/ItemService.java]: fixed

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 4 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: fixed

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 3 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: fixed

File: src/main/java/ru/itis/mappers/ItemMapper.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+import ru.itis.dto.ItemDto;
+import ru.itis.models.Item;
+
+@Mapper
+public interface ItemMapper {
+    ItemMapper ITEM_MAPPER = Mappers.getMapper(ItemMapper.class);
```
Comment (Dimanaux) [inline src/main/java/ru/itis/mappers/ItemMapper.java]: Why was it named as a CONSTANT?

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 9]
```diff
@@ -22,29 +22,30 @@ public WishListController(ItemService itemService, WishListService wishListServi
...
     }
 
     @CrossOrigin
-    @DeleteMapping("/lists/{list-id}")
+    @DeleteMapping("/lists/{listId}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Delete an item")
-    public void deleteItem(@PathVariable("list-id") Long listId, @RequestParam("title") String itemName,
-                           @RequestHeader(name = "AUTH") String token) {
+    public void deleteItem(@PathVariable Long listId, @RequestParam("title") String itemName,
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: Fix code style. You may use `method(param1, param2, param3)` either
```
method(param1,
       param2,
       param3)
```
Don't mix them.

File: src/main/java/ru/itis/dto/ItemDto.java [hunk 10]
```diff
@@ -1,22 +1,13 @@
 package ru.itis.dto;
 
-import lombok.Builder;
 import lombok.Data;
-import ru.itis.models.Item;
+import ru.itis.models.WishList;
 
 @Data
-@Builder
 public class ItemDto {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/dto/ItemDto.java]: The whole idea of DTO is a lame. So mappers are just a crutch

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 11]
```diff
@@ -22,29 +22,30 @@ public WishListController(ItemService itemService, WishListService wishListServi
...
-    public void addNewItem(@PathVariable("list-id") Long listId, ItemForm itemForm, @RequestHeader(name = "AUTH") String token) {
-        itemService.addNewItem(itemForm, listId);
+    public void addNewItem(@PathVariable Long listId, ItemDto itemDto, @RequestHeader(name = "Authorization")
+            String token) {
+        itemService.addNewItem(itemDto, listId);
     }
 
     @CrossOrigin
-    @DeleteMapping("/lists/{list-id}")
+    @DeleteMapping("/lists/{listId}")
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: Thanks, God! You got it!

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 12]
```diff
@@ -22,29 +22,30 @@ public WishListController(ItemService itemService, WishListService wishListServi
     }
 
     @CrossOrigin
-    @GetMapping("/lists/{list-id}")
+    @GetMapping("/lists/{listId}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Get a single wish list")
-    public WishList getWishList(@PathVariable("list-id") Long listId, @RequestHeader(name = "AUTH") String token) {
-//      TODO: if anonym --> ability to reserve an item
+    public WishList getWishList(@PathVariable Long listId, @RequestHeader(name = "Authorization")
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: What a code style heck!

File: src/main/java/ru/itis/mappers/ItemMapper.java (hunk 8 above)
Comment (AinaArd) [reply src/main/java/ru/itis/mappers/ItemMapper.java]: fixed

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 9 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: fixed

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 12 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: fixed

File: src/main/java/ru/itis/services/WishListService.java (hunk 1 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: fixed

File: src/main/java/ru/itis/services/WishListService.java [hunk 13]
```diff
@@ -37,10 +38,12 @@ public WishList addNewWishList(String title, String token) {
     public boolean removeByTitle(String title, String token) {
         Optional<WishList> wishListCandidate = wishListRepository.findByTitle(title);
         if (wishListCandidate.isPresent()) {
-            User currentUser = userService.findUserByToken(token).get();
-            currentUser.getWishLists().remove(wishListCandidate.get());
-            userService.save(currentUser);
-            wishListRepository.delete(wishListCandidate.get());
+            if (userService.findUserByToken(token).isPresent()) {
```
[the remark this answers is not in the document]
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: fixed

File: src/main/java/ru/itis/controllers/ProfileController.java [hunk 14]
```diff
@@ -37,10 +37,10 @@ public ProfileController(WishListService wishListService, UserService userServic
     public ResponseUserDto getProfilePage(@RequestHeader(name = "Authorization") String token) {
         Optional<User> userCandidate = userService.findUserByToken(token);
         if (userCandidate.isPresent()) {
-            return USER_MAPPER.userToResponseUserDto(userCandidate.get());
+            return UserMapper.userToResponseUserDto(userCandidate.get());
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/ProfileController.java]: I like `UserMapper` more than `USER_MAPPER` 👍

File: src/main/java/ru/itis/dto/ItemDto.java (hunk 10 above)
Comment (AinaArd) [reply src/main/java/ru/itis/dto/ItemDto.java]: sometimes it's a good solution

File: src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java (hunk 6 above)
Comment (AinaArd) [reply src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java]: still use

File: src/main/java/ru/itis/services/WishListService.java (hunk 2 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: fixed

**Pull request merged**